crates/core/tedge/src/cli/connect/cli.rs

```diff
@@ -39,25 +43,31 @@ pub enum TEdgeConnectOpt {
 impl BuildCommand for TEdgeConnectOpt {
     fn build_command(self, context: BuildContext) -> Result<Box<dyn Command>, crate::ConfigError> {
         Ok(match self {
-            TEdgeConnectOpt::C8y { is_test_connection } => ConnectCommand {
+            TEdgeConnectOpt::C8y {
+                is_test_connection,
+                offline_mode,
+            } => ConnectCommand {
                 config_location: context.config_location.clone(),
                 config: context.load_config()?,
                 cloud: Cloud::C8y,
                 is_test_connection,
+                offline_mode,
                 service_manager: service_manager(&context.config_location.tedge_config_root_path)?,
             },
             TEdgeConnectOpt::Az { is_test_connection } => ConnectCommand {
                 config_location: context.config_location.clone(),
                 config: context.load_config()?,
                 cloud: Cloud::Azure,
                 is_test_connection,
+                offline_mode: false,
```

Comment:
Why no `offline_mode` for Az and AWS?

Reply:
We don't have to worry about supporting `--offline` for the `tedge reconnect <mapper>` as the command does not support.

If users want to regenerate the bridge configuration, then they can already do so using `tedge refresh-bridges` which will rebuild the existing bridge files without having to test the cloud connection.